lib/room.cpp

```diff
@@ -557,6 +557,23 @@ QString Room::canonicalAlias() const
 
 QString Room::displayName() const { return d->displayname; }
 
+QStringList Room::pinnedEventIds() const {
+    return d->getCurrentState<RoomPinnedEvent>()->pinnedEvents();
+}
+
+QVector< const Quotient::RoomEvent* > Quotient::Room::pinnedEvents() const
+{
+    QStringList events = d->getCurrentState<RoomPinnedEvent>()->pinnedEvents();
+    QVector<const RoomEvent*> pinnedEvents;
+    QStringList::iterator i;
+    for (i = events.begin(); i != events.end(); ++i) {
+        auto timelineItem = findInTimeline(*i);
+        if (timelineItem != historyEdge())
+            pinnedEvents.append(timelineItem->event());
```

Comment:
We could fetch the event if it's not in the timeline with Quotient::GetEventContextJob.

Reply:
We could, but this would quite considerably complicate the logic and make the whole accessor asynchronous. For now I'd say it's an iteration good enough for merging: it allows clients who're up for it to request additional events (and manage their lifecycle), while naive implementations can get a limited amount of data in a very straightforward way. Once jobs in Quotient become proper promise objects, asynchronous access will become easier and `pinnedEvents()` can fetch events before returning (or, as another option, have some "on-demand" loaded vector of event pointers in the same vein `Room::avatar()` works now).